ci/run_one.sh

```diff
@@ -0,0 +1,49 @@
+#!/bin/bash
+
+# Runs shadow's CI for a single configuration.
+# Requires env parameters:
+#   CONTAINER: container-name to use
+#   CC: C compiler to use (cc or clang)
+#   BUILDTYPE: release, debug, or coverage
+#
+# Example:
+#   sudo CONTAINER=centos:8 CC=clang BUILDTYPE=debug ci/run_one.sh
+
+set -euo pipefail
+
+TAG="shadow:${CONTAINER/:/-}-$CC-$BUILDTYPE"
+echo "Running $TAG"
+
+# Build and tag a Docker image witht he given configuration.
+docker build -t "$TAG" -f- . <<EOF
+FROM $CONTAINER
+
+ENV CONTAINER $CONTAINER
+SHELL ["/bin/bash", "-c"]
+
+COPY ci/container_scripts/install_deps.sh /root/install_deps.sh
+RUN . /root/install_deps.sh
```

Comment:
For these steps, is there an advantage to sourcing them rather than running them? I don't think the environment variables carry over between RUNs, so it might be good to remove the `.` from the RUN steps.

Reply:
Ah okay I didn't think of that. Feel free to run them however you think is best, not making them +x is a good reason.